RestaurantReservation.Db/Interfaces/IOrderRepository.cs

```diff
@@ -0,0 +1,16 @@
+﻿using RestaurantReservation.Db.Models;
+
+namespace RestaurantReservation.Db.Interfaces
+{
+    public interface IOrderRepository
+    {
+        Task CreateOrderAsync(Order order);
+        Task DeleteOrderAsync(int orderId);
+        Task<IEnumerable<Order>> GetAllOrdersAsync();
+        Task<Order> GetOrderAsync(int orderId);
+        Task UpdateOrderAsync(Order order);
+        Task<double> CalculateAverageOrderAmountAsync(int employeeId);
+        Task<IEnumerable<Order>> ListOrdersAndMenuItemsAsync(int reservationId);
+        Task<IEnumerable<MenuItem>> ListOrderedMenuItemsAsync(int reservationId);
```

Comment:
Can we use Get instead of List? and what is the difference between the two methods?

Reply:
`**ListOrdersAndMenuItems(ReservationId)**`: This method takes a reservation identifier as a parameter and lists the orders placed on that specific reservation along with the associated menu items.

`**ListOrderedMenuItems(ReservationId)**`: This method takes a reservation identifier as a parameter and finds the menu items ordered in that specific reservation.

Reply:
**`ListOrdersAndMenuItems(ReservationId)`**: takes a reservation identifier as a parameter and lists the orders placed on that specific reservation along with the associated menu items.
**`ListOrderedMenuItems(ReservationId)`**: takes a reservation identifier as a parameter and finds the menu items ordered in that specific reservation.